Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -100,9 +141,36 @@ public override void Close()
             _connectionState = ConnectionState.Closed;
         }
 
+        public override void Close()
+        {
+            SnowflakeDbConnection conn = SnowflakeDbConnectionPool.getConnection(this.ConnectionString, this.Password);
+            if (conn != null)
+            {
+                conn._refCount--;
+            }
+            else
+            {
+                CloseConnection();
+            }
+        }

```

Comment:
This Close() is not actually doing anything turning off this SnowflakeDbConnection. What if users accidentally run multiple time close() on the same connection object?